File: passwd.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <ctime>
#include <fstream>

using namespace std;

// '£' won't print. wonder why?
string char_set1 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!\"$%^&*()-_=+[{]};:'@#~|,<.>/?";
string char_set2 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
string char_set3 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
string char_set4 = "abcdefghijklmnopqrstuvwxyz1234567890";
string char_set5 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
string char_set6 = "abcdefghijklmnopqrstuvwxyz";
string char_set7 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
string char_set8 = "1234567890";
string char_set9 = "custom set";
string characters = "";

string choose_char()
{
  int char_set = 0;
  cout <<"1. "<<char_set1<<endl<<"2. "<<char_set2<<endl<<"3. "<<char_set3<<endl<<"4. "<<char_set4<<endl<<"5. "<<char_set5<<endl<<"6. "<<char_set6<<endl<<"7. "<<char_set7<<endl<<"8. "<<char_set8<<endl<<"9. "<<char_set9<<endl<<"Choose set of characters 1-9: ";
  cin >> char_set;
  switch(char_set){
  case 1:
    return char_set1;
    break;
  case 2:
    return char_set2;
    break;
  case 3:
      return char_set3;
      break;
  case 4:
      return char_set4;
      break;
  case 5:
      return char_set5;
      break;
  case 6:
      return char_set6;
      break;
  case 7:
      return char_set7;
      break;
  case 8:
      return char_set8;
      break;
  case 9:
      
      cout << "Input Characters to use: ";
      cin >> char_set9;
      return char_set9;
      break;
  default:
      return "[!!] NO CHARACTER SET CHOSEN";
  }
}


void generate(string chars)
{
  ofstream myfile;
  myfile.open("list.txt");
  int char_len = chars.length(); int pass_len; int multi; string password; 
  
  cout << "How long is the password?: ";
  cin >> pass_len;

  cout << "How many passwords?: ";
  cin >> multi;
  
  srand(time(0));
  string passwords[multi];
  
  for(int x=0; x<multi; x++) {
    for(int y=0; y<pass_len; y++) {
      int ran_num = (rand () % char_len);
      string ran_char;
      ran_char = chars[ran_num];
      password += ran_char;
    }
    passwords[x] = password;  
    password = "";
    //cout << passwords[x]<< endl;
    myfile << passwords[x] << endl;
  }
  myfile.close();
}


int main()
{

  string characters = choose_char();
  generate(characters);

}

File: passwd.py
#!/bin/python3

import sys
import secrets


char_set1 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!\"£$%^&*()-_=+[{]};:'@#~\|,<.>/?"
char_set2 = "abcdefghijklmnopqrstuvwxyz"
char_set3 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
char_set4 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
char_set5 = "1234567890"
char_set6 = "abcdefghijklmnopqrstuvwxyz1234567890"
char_set7 = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
char_set8 = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"

print("OPTION: characters")
print("1: ", char_set1)
print("2: ", char_set2)
print("3: ", char_set3)
print("4: ", char_set4)
print("5: ", char_set5)
print("6: ", char_set6)
print("7: ", char_set7)
print("8: ", char_set8)
print("9: custom_set")

choice = input("which option of characters do you want to include?: ")
characters = ""

if choice == "1":
	characters = char_set1
if choice == "2":
	characters = char_set2
if choice == "3":
	characters = char_set3
if choice == "4":
	characters = char_set4
if choice == "5":
	characters = char_set5
if choice == "6":
	characters = char_set6
if choice == "7":
	characters = char_set7
if choice == "8":
	characters = char_set8
if choice == "9":
    characters = input("input characters you want to use: ")

length = len(characters)
pass_len = int(input("how long for the password?: "))
multiple = int(input("how many passwords should be created?: "))
passwords = [ ]
mylist = []
while len(mylist) < multiple:
    for j in range(multiple):
        password = ''.join(secrets.choice(characters) for i in range(pass_len-1))
        passwords.append(password)
        #password = ""
        mylist = passwords
        mylist = list(dict.fromkeys(mylist))
	

output = input("print resuts or store in text file? [p/s]: ")
appearance = input("show numbers on side? [y/n]: ")

if output == "p":
    if appearance == "y":
        for i in range(len(mylist)):
            print(i+1,".",mylist[i-1])
    if appearance == "n":
        for i in range(len(mylist)):
            print(mylist[i-1])

if output == "s":
    f = open("passwd.txt","w+")
    if appearance == "y":
        print("[+]Passwords are stored as \'passwd.txt\' in current dirctory[+]")
        for i in range(len(mylist)):
            i_str = i+1
            f.write(str(i+1))
            f.write(".")
            f.write(mylist[i-1])
            f.write('\n')
            f.close
        if appearance == "n":
            print("[+]Passwords are stored as \'passwd.txt\' in current dirctory[+]")
            for i in range(len(mylist)):
                f.write(mylist[i-1])
                f.write('\n')
                f.close

